src/commonMain/kotlin/exchange.dydx.abacus/state/manager/V4StateManagerAdaptor+Transfer.kt

```diff
@@ -453,3 +456,93 @@ internal fun V4StateManagerAdaptor.transferNobleBalance(amount: BigDecimal) {
         }
     }
 }
+
+internal var isCctpWithdraw: Boolean = false
+internal var cctpWithdrawPayload: String? = null
+internal var cctpWithdrawCallback: TransactionCallback? = null
+
+internal fun clearCctpWithdraw() {
+    isCctpWithdraw = false
+    cctpWithdrawPayload = null
+    cctpWithdrawCallback = null
+}
+internal fun V4StateManagerAdaptor.cctpToNoble(
+    state: PerpetualState?,
+    decimals: Int,
+    gas: BigDecimal,
+    callback: TransactionCallback
```

Comment:
Since caller is supplying the callback, can we make sure the callback is invoked on every code path?